client/subcomponents/volunteers/RockUpVolunteerForm/AddVolunteerFormHelper.js

```diff
@@ -3,24 +3,20 @@ import { dispatch, getState } from '../../../store'
 import { setWaiting, clearWaiting } from '../../../slices/waiting'
 import { showError } from '../../../slices/error'
 
-export function addVolunteer(
+export async function addVolunteer(
   volunteer,
   addExtraVolunteer,
   consume = requestor
 ) {
-  const { token } = getState().user
-  dispatch(setWaiting())
-
-  return consume('/volunteers/extras', token, 'post', volunteer)
-    .then((response) => {
-      dispatch(clearWaiting())
-      const newVolunteer = { ...volunteer, ...response.body }
-      addExtraVolunteer(newVolunteer)
-      return null
-    })
-
-    .catch((error) => {
-      dispatch(showError(error.message))
-      return null
-    })
+  try {
+    const { token } = await getState().user
+    await dispatch(setWaiting())
+    const response = await consume('/volunteers/extras', token, 'post', volunteer)
+    await dispatch(clearWaiting())
+    const newVolunteer = { ...volunteer, ...response.body }
+    await addExtraVolunteer(newVolunteer)
+    return response
```

Comment:
i thiiink this function needs to return null but let me know if you think otherwise or it doesn't work

Reply:
Thank you!!